frontend/src/metabase/admin/datamodel/containers/FieldApp.jsx

```diff
@@ -124,26 +122,22 @@ export default class FieldApp extends React.Component {
       fieldId,
       fetchDatabaseMetadata,
       fetchTableMetadata,
-      fetchDatabaseIdfields,
       fetchFieldValues,
     } = this.props;
 
-    // A complete database metadata is needed in case that foreign key is changed
-    // and then we need to show FK remapping options for a new table
-    await fetchDatabaseMetadata(databaseId);
+    await Promise.all([
+      // A complete database metadata is needed in case that foreign key is
+      // changed and then we need to show FK remapping options for a new table
+      fetchDatabaseMetadata(databaseId),
 
-    // Only fetchTableMetadata hydrates `dimension` in the field object
-    // Force reload to ensure that we are not showing stale information
-    await fetchTableMetadata(tableId, true);
+      // Only fetchTableMetadata hydrates `dimension` in the field object
+      // Force reload to ensure that we are not showing stale information
+      fetchTableMetadata(tableId, true),
 
-    // load field values if has_field_values === "list"
-    const field = this.props.metadata.field(fieldId);
-    if (field && field.has_field_values === "list") {
-      await fetchFieldValues(fieldId);
-    }
-
-    // TODO Atte Keinänen 7/10/17: Migrate this to redux/metadata
-    await fetchDatabaseIdfields(databaseId);
+      // always load field values even though it's only needed if
+      // has_field_values === "list"
+      fetchFieldValues(fieldId),
+    ]);
```

Comment:
There were two parts to this change:

1. Remove the explicit fetch for id fields. I think this is covered by the heavyweight pull on database metadata.
2. It seems like the remaining three fetches could be parallelized? The field fetch was contingent on data loaded in the prior two, but it's simpler and probably faster to just load all three at once.